File: fixtures/scale/dart/small/lib/model/document.dart
class Document {
  final List<String> lines;

  Document._(this.lines);

  factory Document.fromText(String text) {
    final lines = text.split('\n').map((l) => l.trimRight()).toList();
    return Document._(lines);
  }

  int get longestLine =>
      lines.isEmpty ? 0 : lines.map((l) => l.length).reduce((a, b) => a > b ? a : b);
}

File: fixtures/scale/kotlin/small/src/textprinter/model/Document.kt
package textprinter.model

class Document private constructor(val lines: List<String>) {

    fun longestLine(): Int = lines.maxOfOrNull { it.length } ?: 0

    companion object {
        fun fromText(text: String): Document =
            Document(text.split("\n").map { it.trimEnd() })
    }
}
